pueue_lib/src/network/socket/unix.rs

```diff
@@ -142,8 +142,19 @@ pub async fn get_listener(settings: &Shared) -> Result<GenericListener, Error> {
             })?;
         }
 
+        // Adjust the umask prior to creating the socket.
+        let old_umask = settings
+            .unix_socket_permissions
+            .map(|permissions| unsafe { libc::umask(!permissions) });
+
         let unix_listener = UnixListener::bind(&socket_path)
-            .map_err(|err| Error::IoPathError(socket_path, "creating unix socket", err))?;
+            .map_err(|err| Error::IoPathError(socket_path.clone(), "creating unix socket", err))?;
+
+        // Reset the umask now that the socket is created.
+        if let Some(permissions) = old_umask {
+            unsafe { libc::umask(permissions) };
+        }
+
```

Comment:
In the case of an error, the umask would be still set to the wrong value.

We cannot use the `?` operator directly, but rather use it after the reset of the umask below.